Jenkinsfile.nightly

```
@@ -107,6 +107,19 @@ pipeline {
                     addStringParam(buildParams, 'MAVEN_DEPENDENCIES_REPOSITORY', getArtifactsRepositoryParam())
                     addStringParam(buildParams, 'MAVEN_DEPLOY_REPOSITORY', getArtifactsRepositoryParam())
 
+                    if (getBuildBranch() != 'master') {
+
+                        /* The OptaPlanner major version is shifted by 7 from the Kogito major version:
+                           Kogito 1.x.y -> OptaPlanner 8.x.y. */
+                        int majorVersionShift = 7
+                        String [] buildBranchSplit = getBuildBranch().split("\\.")
+                        assert buildBranchSplit.length == 3
```

Would it be better to use the shared library's [`parseVersion()`](https://github.com/kiegroup/jenkins-pipeline-shared-libraries/blob/e9a746b8bccdf80aa91d2db2ae2d0ac785bd1e0f/vars/util.groovy#L238-L254) here?

---

Not really: the `parseVersion()` expects (and fails fast otherwise) all the components to be numbers. Here I need to parse the branch name, which is in the following format in the nightly pipeline for other branches than master:
`majorVersion.minorVersion.x`, e.g. `1.0.x`.